src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -343,17 +325,17 @@ public string PropertyPath()
 
             for (int i = 0; i < count - 1; i++)
             {
-                AppendStackFrame(sb, _previous[i]);
+                AppendStackFrame(sb, ref _stack[i]);
             }
 
             if (_continuationCount == 0)
             {
-                AppendStackFrame(sb, Current);
+                AppendStackFrame(sb, ref Current);
             }
 
             return sb.ToString();
 
-            void AppendStackFrame(StringBuilder sb, in WriteStackFrame frame)
+            static void AppendStackFrame(StringBuilder sb, ref WriteStackFrame frame)
```

Comment:
Interesting. Was a defensive copy made prior?

Reply:
Yep, I believe the compiler does this because `WriteStackFrame` is a mutable struct.

Reply:
This is also true for most of the `Converter<T>.Write*` methods that accept an `in T value` parameter whenever `T` is a mutable struct. I was considering marking those `ref` as well, but it's probably overkill given it's meant to optimize for the occasional large mutable struct that gets serialized.